pinot-core/src/main/java/org/apache/pinot/core/plan/FilterPlanNode.java

```diff
@@ -108,23 +114,24 @@ public Map<Predicate, PredicateEvaluator> getPredicateEvaluatorMap() {
   /**
    * H3 index can be applied iff:
    * <ul>
-   *   <li>Predicate is of type RANGE</li>
-   *   <li>Left-hand-side of the predicate is an ST_Distance function</li>
-   *   <li>One argument of the ST_Distance function is an identifier, the other argument is an literal</li>
+   *   <li>Predicate is of type RANGE or EQ</li>
+   *   <li>Left-hand-side of the predicate is an ST_Distance or ST_Contains function</li>
+   *   <li>One argument of the ST_Distance or ST_Contains function is an identifier, the other argument is an
+   *   literal</li>
    *   <li>The identifier column has H3 index</li>
    * </ul>
    */
   private boolean canApplyH3Index(Predicate predicate, FunctionContext function) {
```

Comment:
Suggest having separate check methods for `ST_Distance` and `ST_Contains` to prevent mix-match (e.g. `ST_Distance` with `EQ` is not supported), and also avoid the extra check after the method returns `true`

Reply:
sure. updated.